Added task to copier to warn that template is deprecated and updated …

lempira opened the pull request:
## Proposed Changes
- Updated the README 
- Added a task before the init finalizes to print the deprecation warning and fail the init

lempira committed:
Added task to copier to warn that template is deprecated and updated README

lempira commented on copier.yaml:
```diff
@@ -1,6 +1,10 @@
 _templates_suffix: ".jinja"
 
+_tasks:
+  - "python -c \"print('WARNING: This template has been deprecated. Beaker for Algorand Smart Contract development is no longer supported.\\nPlease use Algorand Python instead. The Algorand Python template can be found at https://github.com/algorandfoundation/algokit-python-template')\""
```
This tasks runs before the init finishes but the files are still copied. Is there a way to run the tasks before prompting for the questions?

lempira committed:
chore: deprecated tests and created a test for a failed init

aorumbayev committed:
chore: deprecate beaker template with explicit question + warning & exit task

aorumbayev reviewed (approved):
Pre approving, please see my slightly different approach here first https://github.com/algorandfoundation/algokit-beaker-default-template/pull/77

lempira committed:
chore: changed working dir before deleting template dir and added more err handling

lempira commented:
> Pre approving, please see my slightly different approach here first #77

@aorumbayev I saw your comment about testing it with the windows binary and I was finding that it wasn't deleting the template dir if you chose not to proceed after the deprecation warning. I added additional error logging and I am getting this error `Failed to clean up C:\Users\lempi\Documents\algokit-init-test\asdf: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\lempi\\Documents\\algokit-init-test\\asdf\\projects\\asdf'` It looks like another file is open so it fails on deleting `asdf` template. I thought it could have been due somehow the template being the workdir so I changed the dir before deleting but I got the same error. I was then think about using something like psutil to force delete files that might be open but was wondering if you had any ideas before I go down that route.

aorumbayev commented:
@lempira i see, yeah so this seems to be caused by the fact that the script running the deletion of the folder is located within that folder. There are several workarounds for this depending on whether you want to go an extra mile for this edge case or not:
- Recursively delete the contents of project folder instead of of running rmtree, the script can be self removed at the very end of script execution (if windows is the running platform) -> user will be left with an empty folder but would still need to cleanup the folder OR you can then run an extra platform specific task https://copier.readthedocs.io/en/stable/configuring/#tasks to cleanup that folder separately. 
- Keeping the post_init.py script execution as a unix specific command (by leveraging platform specific tasks https://copier.readthedocs.io/en/stable/configuring/#tasks) and writing a windows specific inline bash script to remove the directory 
- Handling the exception and printing out the instruction to user to manually remove the folder

lempira committed:
chore: Removed other exceptions. Kept generic exception

aorumbayev commented:
@lempira yeah makes sense, thats probably the most pragmatic, the windows binary is a fairly specific edge case not worth going overboard 👍
